src/viam/app/data/client.py

```diff
@@ -66,19 +71,62 @@ async def delete_binary_data_by_ids(self, file_ids: Optional[List[str]]) -> None
         raise NotImplementedError()
 
     async def add_tags_to_binary_data_by_file_ids(self, file_ids: Optional[List[str]], tags: Optional[List[str]]) -> None:
-        raise NotImplementedError()
+        """
+        Add tags to binary data using file ID's
 
-    async def add_tags_to_binary_data_by_filter(self, filter: Optional[Filter], tags: Optional[List[str]]) -> None:
-        raise NotImplementedError()
+        Args:
+            file_ids (List[str]): List of file ID's specifying binary data to modify
+            tags (List[str]): List of tags to add to specified binary data
+        """
+        request = AddTagsToBinaryDataByFileIDsRequest(file_ids=file_ids, tags=tags)
+        await self._data_client.AddTagsToBinaryDataByFileIDs(request, metadata=self._metadata)
 
-    async def remove_tags_from_binary_data_by_file_ids(self, file_ids: Optional[List[str]], tags: Optional[List[str]]) -> None:
-        raise NotImplementedError()
+    async def add_tags_to_binary_data_by_filter(self, tags: Optional[List[str]], filter: Optional[Filter] = Filter()) -> None:
+        """
+        Add tags to binary data using a filter
 
-    async def remove_tags_from_binary_data_by_filter(self, filter: Optional[Filter], tags: Optional[List[str]]) -> None:
-        raise NotImplementedError()
+        Args:
+            filter (Filter): Filter specifying binary data to modify
+            tags (List[str]): List of tags to add to specified binary data
+        """
+        request = AddTagsToBinaryDataByFilterRequest(filter=filter, tags=tags)
+        await self._data_client.AddTagsToBinaryDataByFilter(request, metadata=self._metadata)
 
-    async def tags_by_filter(self, filter: Optional[Filter]) -> List[str]:
-        raise NotImplementedError()
+    async def remove_tags_from_binary_data_by_file_ids(self, tags: Optional[List[str]], file_ids: Optional[List[str]] = Filter()) -> None:
+        """
+        Remove tags from binary data using file ids
+
+        Args:
+            file_ids (List[str]): List of file ID's specifying binary data to modify
+            tags (List[str]): List of tags to remove from specified binary data
```

Comment:
Looks like we're going to run into the `Optional` vs `Required` question on all of these.

If these truly are optional, the docstring should mention what the behavior is when left empty.

Reply:
See above